File: api/trends.js
import fetch from "node-fetch";

export default async function handler(req, res) {
  const keyword = req.query.keyword || "TSLA";
  const url = `https://trends.google.com/trends/api/explore?hl=en-US&tz=-480&req={"comparisonItem":[{"keyword":"${keyword}","geo":"US","time":"now%207-d"}],"category":0,"property":""}`;

  try {
    const response = await fetch(url);
    const text = await response.text();
    res.status(200).json({ keyword, score: text.length });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}

File: api/trends.py
from pytrends.request import TrendReq
from urllib.parse import parse_qs
import json

def handler(request):
    query = parse_qs(request.query)
    keyword = query.get("keyword", [""])[0]

    if not keyword:
        return {
            "statusCode": 400,
            "body": json.dumps({"error": "Missing keyword"})
        }

    pytrends = TrendReq(hl='en-US', tz=360)
    pytrends.build_payload([keyword], cat=0, timeframe='now 7-d', geo='', gprop='')
    data = pytrends.interest_over_time()

    if data.empty:
        score = 0
    else:
        score = int(data[keyword].iloc[-1])

    return {
        "statusCode": 200,
        "body": json.dumps({"score": score})
    }
